src/window.rs

```diff
@@ -156,6 +158,36 @@ impl Window {
             .or_else(|_| create_gl_window(window_builder, &event_loop, true))?;
         window.show();
 
+        #[cfg(not(any(windows, target_os = "macos")))]
+        {
+            if window_config.decorations() == Decorations::Transparent && event_loop.is_wayland() {
+                let bg = colors.primary.background;
+                let mut scale = 382.5 / (f32::from(bg.r) + f32::from(bg.g) + f32::from(bg.b));
+                scale /= 3.;
```

Comment:
This will be infinity when the background is completely black, casting it to `u8` then sets it to 0, leading to the background not getting scaled.